ansible/roles/mn_fund_collateral/tasks/main.yml

```diff
@@ -35,9 +35,9 @@
   ansible.builtin.include_role:
     name: generate_blocks
   vars:
-    num_blocks: 0
-    balance_needed: "{{ (masternode_names | length - collateral_ok_count | int) * (funding_amount | int / 100000) | int }}"
-  when: not collateral_all_ok and (faucet_balance.stdout | int < balance_needed | int)
+    generate: balance
+    target_balance: "{{ (masternode_names | length - collateral_ok_count | int) * (funding_amount | int / 100000) | int }}"
+  when: not collateral_all_ok and (faucet_balance.stdout | int < target_balance | int)
```

Comment:
Don't you have such check already inside?

Reply:
Where? You mean [this check](https://github.com/dashpay/dash-network-deploy/pull/434/files#diff-c83d51b081e08f1a45847f902e089cf58b36a0c879981e56e1bbd339d5e6bd52R48)?

Reply:
Yep

Reply:
One check is whether it is necessary to enter the `generate_blocks` function, the other check is when to stop generating the blocks (after determining it was necessary to start in the first place).

There's no need to start and stop the systemd mining service just to run the balance check.